public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ConfigPublicDashboard/SettingsBar.tsx

```diff
@@ -0,0 +1,45 @@
+import { css } from '@emotion/css';
+import React, { useCallback, useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+
+import { SettingsBarHeader } from './SettingsBarHeader';
+
+export interface Props {
+  title: string;
+  headerElement?: React.ReactNode | ((props: { className?: string }) => React.ReactNode);
+  children: React.ReactNode;
+}
+
+export function SettingsBar({ children, title, headerElement }: Props) {
+  const styles = useStyles2(getStyles);
+  const [isContentVisible, setIsContentVisible] = useState(false);
+
+  const onRowToggle = useCallback(() => {
+    setIsContentVisible(!isContentVisible);
+  }, [isContentVisible, setIsContentVisible]);
```

Comment:
I think it's better to use prevState for this kind of things, so there's no need to use useCallback neither.

```suggestion
 const onRowToggle = () => {
    setIsContentVisible((prevState) => !prevState);
  };
```